src/SimpleWebRequest.ts

```diff
@@ -659,38 +666,40 @@ export class SimpleWebRequest<T> extends SimpleWebRequestBase {
 
         if (this._xhr && this._xhr.readyState === 4 && ((statusCode >= 200 && statusCode < 300) || statusCode === 304)) {
             // Happy path!
-            const resp: WebResponse<T> = {
+            const resp: WebResponse<ResponseBody, RequestOptions> = {
                 url: this._xhr.responseURL || this._url,
                 method: this._action,
-                requestOptions: this._options,
+                requestOptions: this._options as RequestOptions,
```

Comment:
Add generics to the base class for the options type like `<TOptions extends WebRequestOptions>` and use it for `_options: TOptions`, instead of these casts for `this._options as RequestOptions`.